internal/proxyservice/proxy_service.go

```diff
@@ -34,8 +36,51 @@ type proxyServices struct {
 	runningServices []internal.Service
 }
 
+func (s *proxyServices) validateRequestedPluginsExist() error {
+	allPlugins := []string{"ssh", "ssh-agent"}
+	for pluginName := range s.availPlugins.TCPPlugins() {
+		allPlugins = append(allPlugins, pluginName)
+	}
+	for pluginName := range s.availPlugins.HTTPPlugins() {
+		allPlugins = append(allPlugins, pluginName)
+	}
+
+	s.logger.Infof(
+		"Validating services against available plugins: %s",
+		strings.Join(allPlugins, ","),
+	)
+
+	allPluginsMap := map[string]bool{}
+	for _, p := range allPlugins  {
+		allPluginsMap[p] = true
+	}
```

Comment:
I had to do a minor double-take to figure out that `allPluginsMap` existed solely so that we don't have to have a nested loop within the loop below to check for existence.  This is a good reason to have it, but a small comment here would help readers know what it's for without having to read the code below.  Something like:

```go
// Convert available plugins to a map, so that we can check for their
// existence in the loop below using a map lookup rather than a nested
// loop, which would be required otherwise.